+neuron/Object.m

```diff
@@ -0,0 +1,163 @@
+classdef Object < dynamicprops
+% Neuron Object Class
+
+    properties (SetAccess=protected, GetAccess=public)
+        objtype         % Neuron Object type
+        obj             % C++ Neuron Object.
+        attr_list       % List of attributes of the C++ object.
+        mt_double_list  % List of methods of the C++ object, returning a double.
+        mt_object_list  % List of methods of the C++ object, returning an object.
+        mt_string_list  % List of methods of the C++ object, returning a string.
+    end
+    
+    methods
+
+        function self = Object(objtype, obj)
+        % Initialize Object
+        %   Object(objtype, obj) constructs a Matlab wrapper for Neuron 
+        %   Object obj of type objtype
+
+            self = self@dynamicprops;
+            if clib.neuron.isinitialized()
+                self.objtype = objtype;
+                self.obj = obj;
+
+                % Get method list.
+                method_str = clib.neuron.get_class_methods(self.objtype);
+                method_list = split(method_str, ";");
+                method_list = method_list(1:end-1);
+
+                % Add dynamic properties.
+                for i=1:length(method_list)
+                    method = split(method_list(i), ":");
+                    if (method(2) == "311")
+                        self.attr_list = [self.attr_list method(1)];
+                        p = self.addprop(method(1));
+                        self.(method(1)) = clib.neuron.get_pp_property(self.obj, method(1));
```

Comment:
Is it even needed to get the property value at this point?
Since actually accessing the property goes through the overloaded subsref, that will then (correctly) do a get_pp_property at that time?

Reply:
This is just so that if you initialize the Object, and display it in MATLAB, it already 'knows' what the Object property values are.

Reply:
Added a comment to the code for clarification.